File: 0083 - Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        original_head = head
        while head is not None:
            block = head.next
            while block is not None and block.val == head.val:
                block = block.next
            head.next = block
            head = block
        return original_head

File: 0083 - Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head) return head;
        ListNode* left = head;
        ListNode* right, to_delete;
        while(left){
            right = left->next;            
            while(right && right->val == left->val){
                right = right->next;
            }
            left->next = right;
            left = right;
        }
        return head;
    }
};